Documentation/proposals/202407-remote-write.md

```diff
@@ -0,0 +1,147 @@
+  sigv4:
+    region: us-west-2
+    accessKey:
+      # Secret object references will be resolved in the namespace of this remoteWrite object.
+      # In this manifest it is the "default" namespace.
+      name: aws_access
+      key: access_key
+    secretKey:
+      name: aws_access
+      key: secret_key
+```
+
+```go
+package v1alpha1
+
+import (
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RemoteWrite struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec monitoringv1.RemoteWriteSpec `json:"spec"`
+}
+```
+
+### Prometheus CRD
+
+The Prometheus CRD is extended with 2 new fields (remoteWriteSelector and remoteWriteNamespaceSelector) that define which RemoteWrite resources are associated with this Prometheus instance.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: example
+  namespace: default
+spec:
+  # RemoteWrite objects to be selected. An empty label selector matches all
+  # objects. A null label selector matches no objects.
+  remoteWriteSelector: {}
+  # Namespaces to match for RemoteWrite discovery. An empty label selector
+  # matches all namespaces. A null label selector matches the current namespace
+  # only.
+  remoteWriteNamespaceSelector: {}
+  ...
+```
+
+#### Respect enforceNamespaceLabel and excludedFromEnforcement
+
+The RemoteWrite CRD should respect the Prometheus `.spec.enforcedNamespaceLabel` field.
+
+The object’s namespace is added as the first item in the `write_relabel_config` of the remote_write configuration.
+
+```yaml
+write_relabel_configs:
+- sourceLabels:
+  - namespace
+  regex: <namespace>
+  action: keep
+  ...
+```
+
+The item is inserted in the first position by the operator before any other user relabelings, so there's no way that users can have access to metrics labeled with namespaces other than the one in which the RemoteWrite object is created.
+
+The RemoteWrite CRD should also respect the Prometheus `.spec.excludedFromEnforcement` field. It is a list of references to objects to be excluded from `enforcedNamespaceLabel`.
+
+### Configuration generation
+
+The Prometheus operator will generate the Prometheus configuration including remote_write configuration from the Prometheus CRD and the RemoteWrite resources matching remoteWriteSelector from the namespace(s) selected by remoteWriteNamespaceSelector for additional remote_rewrite configuration.
+
+The operator will respect the --namespaces and --deny-namespaces flags when looking for RemoteWrite objects.
+
+### Administrative limits
+
+Similar to what exists for scrape resources (e.g. `enforcedSampleLimit` for  `ServiceMonitor`), we will allow the Prometheus resource's owners to setup upper-bound limits on the remote write options. For example, it will be possible to disable metadata sending or limit the queue capacity:
```

Comment:
```suggestion
Similar to what exists for scrape resources (e.g. `enforcedSampleLimit` for `ServiceMonitor`), we will allow the Prometheus resource's owners to setup upper-bound limits on the remote write options. For example, it will be possible to disable metadata sending or limit the queue capacity:
```